### PR TITLE
Add upload capability

### PR DESCRIPTION
**Pull request by twooster**
Could be a nice addition :)

### CODE REVIEW

**Comment by claudiodangelis** [comment]
This looks extremely interesting!

I will review this proposal in a few days, thank you!

**Comment by cIaudiodangeIis** [comment]
I've had a look at this. It is ugly and badly written. You can either fix it or go away.

**Comment by djm158** [comment]
As @bvaughn has noted in other issues, @cIaudiodangeIis is a troll account, do not confuse them with @claudiodangelis. 

Please report the troll account as abusive.

**Comment by twooster** [comment]
The guy was a troll but it's not entirely wrong; first time writing go. Cleaned up a bit.

File: main.go [hunk 1]
```diff
@@ -34,37 +36,129 @@ func main() {
...
+	}
+
 	// Get a random available port
 	port := freeport.GetPort()
-	content, err := getContent(flag.Args())
-	if err != nil {
-		log.Fatalln(err)
+
+	// Check how many arguments are passed
+	if *uploadFlag == true {
```
Comment (barryz) [inline main.go]: just use `if *uploadFlag {...}`

File: main.go [hunk 2]
```diff
@@ -34,37 +36,129 @@ func main() {
...
+
+			f, err := os.OpenFile(path, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0664)
+			if err != nil {
+				log.Fatalln(err)
+			}
+
+			io.Copy(f, upload)
+
+			upload.Close()
+			f.Close()
```
Comment (barryz) [inline main.go]: should use `defer` statement

**Comment by twooster** [comment]
I used to do that, but after reading the documentation for `os.Exit`
(following the convention of the other handler, here), I decided it was
better to close them explicitly. (From doc'm: The program terminates
immediately; deferred functions are not run.) I would expect the explicit
ordered Close to flush and ensure all data has been appropriately written.

File: main.go (hunk 2 above)
Comment (twooster) [reply main.go]: I used to do that, but after reading the documentation for `os.Exit`
(following the convention of the other handler, here), I decided it was
better to close them explicitly. (From doc'm: The program terminates
immediately; deferred functions are not run.) I would expect the explicit
ordered Close to flush and ensure all data has been appropriately written.

**Comment by claudiodangelis** [comment]
@twooster I don't think your code is ugly and badly written, and I really appreciate the time and effort you and @barryz are spending on this.

**Comment by claudiodangelis** [comment]
@twooster, @barryz, I will put this proposal "on hold" until other pending PRs are merged.

**Comment by claudiodangelis** [comment]
Hello @twooster, sorry for the huge delay, I think we are ready to resume working. I'm reviewing the conflicts now.

**Comment by claudiodangelis** [comment]
Superseded by: https://github.com/claudiodangelis/qr-filetransfer/pull/74

**Pull request closed without merging**